galter_subjects_utils/keeptrace.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 Northwestern University.
+#
+# galter-subjects-utils is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""KeepTrace."""
+
+from dataclasses import dataclass
+
+
+@dataclass
+class KeepTrace:
+    """Keeps trace of subject at field in record using template."""
+
+    field: str  # dotted path to field
+    template: str
+    yes = "Y"
+    no = "N"
+    keep_trace_key = "keep_trace"
+
+    @classmethod
+    def mark(cls, ops_data, yes_logic):
+        """Mark ops_data in-place to keep trace or not depending on yes_logic.
+
+        :param ops_data: operation data
+        :type ops_data: List[dict] (list of subject op data)
+        :param yes_logic: function to determine if op_data should be marked
+        :type yes_logic: function(op_data) -> Bool
+        """
+        for op_data in ops_data:
+            op_data[cls.keep_trace_key] = cls.yes if yes_logic(op_data) else cls.no  # noqa
+
+    def should_trace(self, op_data):
```

Comment:
This could be placed inside `trace`, but I think this makes the check more explicit for a reader of the code where `should_trace` + `trace` are used. Welcome thoughts on that though.

Reply:
I go back and forth on wrapping comparisons in functions, but I generally agree with you that putting a name to it makes it easier to read and reason about. I think it fits with other decisions you have made in the code so I think this is a good approach 👍